### ISSUE TITLE
CI error: signing certs invalid

### ISSUE THREAD

**Issue by mmaietta** [electron-userland/electron-builder#5635]
The CI is currently showing broken tests for MacOS signing. I was unable to repro locally in using my personal certs, codesign completely properly.
I added `DEBUG='electron-builder'` to the TracisCI config and was able to isolate the error to expired certificates.

```
  • skipped macOS application code signing  reason=cannot find valid "Developer ID Application" identity or custom non-Apple code signing certificate, see https://electron.build/code-signing allIdentities=
		1) C9A30EAD8807BE62058F94B4385B53A7BE3C9392 "Mac Developer: Vladimir Krivosheev (744739DJ4Y)" (CSSMERR_TP_CERT_EXPIRED)
		2) 44DE023F1769CEF2068AA042F595EE95D381AC6C "Developer ID Installer: Vladimir Krivosheev (X8C9Z9L4HW)" (CSSMERR_TP_CERT_EXPIRED)
		3) 2ABC71002FE96EBE6536C4AECC88013D6EB50443 "Developer ID Application: Vladimir Krivosheev (X8C9Z9L4HW)" (CSSMERR_TP_CERT_EXPIRED)
		4) 37CAC6B3B3E04B7FA1C47DDF75B204A9C27350B8 "3rd Party Mac Developer Installer: Vladimir Krivosheev (X8C9Z9L4HW)" (CSSMERR_TP_CERT_EXPIRED)
		5) 42D1E4025947845C59046BB7196279B9D9F3426A "3rd Party Mac Developer Application: Vladimir Krivosheev (X8C9Z9L4HW)" (CSSMERR_TP_CERT_EXPIRED)
		   5 identities found
                                              
		Valid identities only
		   0 valid identities found
```

@develar can you please regenerate Apple Dev Certs for the unit tests?

I'm unable to update the certificates myself for this as my Apple dev account is through my work. I believe you'd have the master CSC_LINK_PASSWORD for the CI for reuse as well.

**Comment by mmaietta** [comment]
@develar I'm also receiving similar errors for Windows tests on CI. Just need to add another node `os: windows` with `TEST_FILES=windows/*`

```
 FAIL  src/windows/winCodeSignTest.ts (6.001 s)
  × sign nested asar unpacked executables (2921 ms)

    Error: expect(received).toContain(expected) // indexOf

    Expected substring: "Unrecognized file type"
    Received string:    "Exit code: 1. Command failed: C:\\Users\\travis\\AppData\\Local\\electron-builder\\Cache\\winCodeSign\\winCodeSign-2.6.0\\windows-10\\x64\\signtool.exe sign /t http://timestamp.digicert.com /f C:\\Users\\travis\\AppData\\Local\\Temp\\et-bc3522cd7009a553069bd91382fdf583\\t-Pobgxj\\1.p12 /d Test App ßW /du http://foo.example.com /debug C:\\Users\\travis\\AppData\\Local\\Temp\\et-bc3522cd7009a553069bd91382fdf583\\t-Pobgxj\\test-project-0\\dist\\win-unpacked\\Test App ßW.exe
    SignTool Error: The specified PFX password is not correct.·

 FAIL  src/windows/squirrelWindowsTest.ts (15.375 s)
  × Squirrel.Windows (2986 ms)
  × artifactName (11145 ms)

  ● Squirrel.Windows
    Exit code: 1. Command failed: C:\Users\travis\AppData\Local\electron-builder\Cache\winCodeSign\winCodeSign-2.6.0\windows-10\x64\signtool.exe sign /t http://timestamp.digicert.com /f C:\Users\travis\AppData\Local\Temp\et-bc3522cd7009a553069bd91382fdf583\t-5fXZbf\1.p12 /d Test App ßW /du http://foo.example.com /debug C:\Users\travis\AppData\Local\Temp\et-bc3522cd7009a553069bd91382fdf583\t-5fXZbf\test-project-0\dist\win-unpacked\Test App ßW.exe
    SignTool Error: The specified PFX password is not correct.
    
    SignTool Error: The specified PFX password is not correct.
```

**Comment by introkit** [comment]
+1, very tired of this

**Comment by mmaietta** [comment]
@develar any chance you'd be able to tackle this soon?
I'd love to be able to get all tests passing again as it's hard to track offending commits post-merge to master. I'm also able to add a windows node to our TravisCI which would be awesome to activate our coverage on many of those windows-specific tests.
In both cases, the signing certs are needing updates. 😕

**Comment by develar** [comment]
Bloody Apple blocked my developer account because I do not have answers to *** security questions. Even if I know my  password and has access to e-mail, it is not enough to unblock account.

So, will be great if you will generate some kind of test certs.

**Comment by mmaietta** [comment]
Oomph! That's no good at all.

I'm blocked on the developer portal, seems they don't even let you look at certs or revoke em if your membership isn't active. I'll have to renew that.

Where can we generate a cert for Windows?

**Comment by develar** [comment]
> Where can we generate a cert for Windows?

For Windows you cannot get cert easily. And you don't have to. Just generate some code signing cert yourself and it will be enough to test code signing.
<img width="501" alt="Screenshot 2021-03-17 at 08 01 59" src="https://user-images.githubusercontent.com/350686/111427892-127a9680-86f7-11eb-9dd5-a34a5039aee8.png">

Maybe even for macOS self-signed code signing certificate will be enough, but I doubt.

**Comment by mmaietta** [comment]
Quick update. I got mac working with a Mac Development cert I think. Need to fix one mac test AFAICT

Will continue working on the windows cert, but also trying to clean up the tests to make sure I'm solving the correct problems.

**Comment by mmaietta** [comment]
Mac certs are working correctly.

But I can't for the life of me figure out the windows signing issues...
The env var is set with the password, but then the test script says the CSC_KEY_PASSWORD isn't set at all.

Soooo....shot in the dark here in case anyone else stumbles into this issue/ticket, but does anyone know if there a special config required for windows build nodes?

**Comment by deathlyrage** [comment]
I use these

WIN_CSC_LINK=file://Signing/file.pfx
WIN_CSC_KEY_PASSWORD=

**Comment by mmaietta** [comment]
@develar all tests have been fixed in #5806. Can you please cut a new release version? 🙂
